lib/internal/v8_prof_polyfill.js

```diff
@@ -26,40 +26,40 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 // Node polyfill
-var fs = require('fs');
-var os = {
+const fs = require('fs');
+const cp = require('child_process');
+const os = {
   system: function(name, args) {
     if (process.platform === 'linux' && name === 'nm') {
       // Filter out vdso and vsyscall entries.
-      var arg = args[args.length - 1];
+      const arg = args[args.length - 1];
       if (arg === '[vdso]' ||
           arg == '[vsyscall]' ||
           /^[0-9a-f]+-[0-9a-f]+$/.test(arg)) {
         return '';
       }
     }
-    return require('child_process').execFileSync(
-        name, args, {encoding: 'utf8'});
+    const cmd = name + ' ' + args.join(' ');
+    return cp.execSync(cmd, {encoding: 'utf8'});
```

Comment:
`.execSync()` executes the command as `/bin/sh -c '<cmd> <args>'` (`cmd.exe /s /c "<cmd> <args>"` on Windows) whereas `.spawnSync()` simply calls execve().  Guess I was wrong saying it takes care of shell escaping because it doesn't need to escape anything.

You could wrap that bit of shell script in a `sh -c` or `bash -c` below.  Longer term, it might be nice to move `normalizeExecArgs()` from `lib/child_process.js` to `lib/internal/child_process.js` so you can use it here.  It's possible some more work needs to be done on the internal module system to make that work, though.

I'll leave it up to you if you want to change this line.